source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -497,7 +510,82 @@ You can configure your environment by updating the previous file or by defining
     cas.logout.url=${cas.baseURL}/logout?url=${geonetwork.https.url}/
 
+.. _authentication-keycloak:
+
+Configuring Keycloak
+----------------------
+
+`Keycloak <https://keycloak.org>`_ is a software solution to facilitate storage of authentication details, user federation, identity brokering and social login.
+GeoNetwork can be set up to use a keycloak instance for authentication. 
+
+Install keycloak from its instructions or use this example setup in docker
+https://www.keycloak.org/getting-started/getting-started-docker
+
+Keycloak details are defined via environment variables
+
+    .. code-block:: text
+
+        KEYCLOAK_AUTH_SERVER_URL={keycloak url}
+        KEYCLOAK_REALM={realm name}
+        KEYCLOAK_RESOURCE={client name}
+        KEYCLOAK_SECRET={client secret}
+        KEYCLOAK_DISABLE_TRUST_MANAGER={true|false}
+
+You can setup more advance keycloak settings by editing the file
+:file:`WEB-INF/config-security/keycloak.json`
+
+Geonetwork client URL configuration
+===================================
+
+Ensure that when you configure your client that you setup the valid redirect uris to your geonetwork installation.
+i.e. https://localhost:8443/geonetwork/\*. If this is not setup correctly you may get and error indicating that a wrong redirect uri was supplied.
+Also if wanting to test the client backchannel logout then ensure that the admin URL is also set to the geonetwork installation.
+
+Sample user/role/group setup
+============================
+
+Sample Role setup
+`````````````````
+
+In your client role settings (clients -> myclient -> roles). Add the following roles
+
+    .. code-block:: text
+
+        Administrator
+        RegisteredUser
+        Guest
+        sample:UserAdmin
+        sample:Reviewer
+        sample:Editor
+        sample:RegisteredUser
+
+Sample Group configuration
+``````````````````````````
+
+#. Go to keycloak groups (left menu).
+#. Create a new group called "Administrator"
+#. Edit the group. Go to Role Mappings -> Client Roles (myclient) -> select the administrator roles and click on "Add selected"
+   Any user part of the Administrator group will not be an administrator for the myclient applicaton.
+
+Sample User configuration
+`````````````````````````
+
+#. Go to keycloak users (left menu)
+#. Add user if not already done so. Then go to that user.
+#. Go to role Mappings -> Client Roles (myclient) -> select the available roles to be applied and click on "Add selected"
+   or go to Groups -> Available Groups -> Click on the Administrator Group and then click on "Join"
+
+Single Signout
+``````````````
+
+If setting up Single Signout, you may need to ensure that you have proper certificates for keycloak and the geonetwork application.
+And ensure the truststore has been setup https://www.keycloak.org/docs/4.8/server_installation/index.html#_truststore
+You will need to ensure that the Admin URL is configured as identified in the URL Configurations above.
```

Comment:
"identified in the URL Configurations above."

This was based on an image and I don't see an image reference in this documentation.

So maybe just say.

`You will need to ensure that the Admin URL is configured in the Keycloak client configuration.`

Reply:
I removed the section, because it seemed specific to a situation, please suggest again if you think is usefull to all.